buildSrc/src/main/kotlin/plugins/RootProjectVersionPlugin.kt

```diff
@@ -0,0 +1,42 @@
+package plugins
+
+import buildReceiptName
+import extensions.VersionInfo
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.*
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.util.Properties
+
+const val incomingBuildReceiptLocation = "incoming-distributions/$buildReceiptName"
+
+open class RootProjectVersionPlugin : Plugin<Project> {
```

Comment:
Could that be a precompiled script plugin?

Reply:
Yeah. I converted it.